react-native-purchases-ui/src/index.tsx

```diff
@@ -62,6 +62,7 @@ export type PresentPaywallIfNeededParams = PresentPaywallParams & {
 
 export interface PaywallViewOptions {
   offering?: PurchasesOffering | null;
+  fontFamily?: String | null;
```

Comment:
As you mentioned, we might want to have this be a map of an enum representing styles (normal, bold,...) to fontFamily names. Wdyt? Let us know if you need any help on this!

Reply:
Sorry, Ive been busy last week. I will pick this back up this week!
Will definitely let u guys know if help is needed!

Reply:
@tonidero 
here's my current findings, extra eyes on this will be very appreciated

I have the following xml for my font family
```xml
<font-family xmlns:app="http://schemas.android.com/apk/res-auto">
  <font
        app:font="@font/montserrat"
        app:fontStyle="normal"
        app:fontWeight="500" />
  <....> etc
</font-family>
```
And I add them on oncreate of MainActivity

```kotlin
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(null);
        ReactFontManager.getInstance().addCustomFont(this, "Montserrat", R.font.montserrat_family);
    }
```

The problem that I'm running into is that ReactFontManager tries to decide which typeface (bold, italics, light) for me.
```java
  public Typeface getTypeface(String fontFamilyName, TypefaceStyle typefaceStyle, AssetManager assetManager) {
        if (this.mCustomTypefaceCache.containsKey(fontFamilyName)) {
            return typefaceStyle.apply((Typeface)this.mCustomTypefaceCache.get(fontFamilyName));
        }
```. Like so.
Since purchases-android only accepts a single CustomFontProvider with a single FontFamily (as it should). It looks like I cannot add change the typefaces with bold, italics etc even with enumeration.

Reply:
Hmm right... I'm not sure but I believe that in RN, you need to specify the fonts separately (not as a font family), then running `npx react-native-asset` would add those font assets to the android/ios projects. I believe the API in the RN side, would need to be a map of style to font names instead (or preferrably an object/interface), and create the font family programmatically in the android side. So I'm thinking something like:
```
interface FontProvider {
   readonly regularFontName: string | null;
   readonly italicFontName: string | null;
   readonly boldFontName: string | null;
}
```
and in presentPaywall, have an optional parameter that receives one of these objects. Wdyt? cc @vegaro for thoughts

Reply:
@tonidero 
yeah I'm lacking knowledge on how https://github.com/unimonkiez/react-native-asset is able to add these fonts to android side of RN. Maybe through merger.xml.
But yeah in order to fix this quickly maybe we can add those values to font provider

Reply:
Hey @Jjastiny,
So I haven't taken a deep look at this, but I think `npx react-native-asset` adds the fonts to the `assets` folder in the android side (probably just copying them?).

One option I was thinking about is, in the native side, to use a constructor for compose's `Font` like [this](https://developer.android.com/reference/kotlin/androidx/compose/ui/text/font/package-summary#Font(kotlin.String,android.content.res.AssetManager,androidx.compose.ui.text.font.FontWeight,androidx.compose.ui.text.font.FontStyle,androidx.compose.ui.text.font.FontVariation.Settings)) which can actually create the font from font files in the assets folder. Once we have a list of `Font`, we can create a `FontFamily`. This relies on the asset being copied to the assets folder, which might change at some point... But seems like the easier way? I haven't researched into how this would be done in iOS yet.

If we do it like this, we would not need to use `ReactFontManager.createAssetTypeface()` I believe?

(I haven't looked too deeply at this so I might be missing something!)

Reply:
@tonidero 
yeah it copies it to the assets folder and ReactFontManager picks it up using the createAssetTypeface.
gotcha, to clarify you want to create a single font family by getting the fonts from the assets folder. 
Yeah we would technically be "re-engineering" what the ReactFontManager is doing, but it does seem like the easiest way to solve this.

For iOS, we can just get the font by using the fontName, without having to specify a style like we do in Android. 
I tested this locally with yalc and it looks like iOS was good, but Android was missing bold text. Lemme share a screenshot for you to see  :)

Reply:
<img width="400" alt="Screenshot 2024-02-22 at 3 57 47 AM" src="https://github.com/RevenueCat/react-native-purchases/assets/6516487/3abf28e4-b57c-4ce8-ae64-cb26d9324652">

This is with the changes in this branch. the bold font shows up perfectly fine on ios

Reply:
<img width="756" alt="Screenshot 2024-02-22 at 5 04 33 AM" src="https://github.com/RevenueCat/react-native-purchases/assets/6516487/ff2290c3-8a7f-4de1-a7ea-62f573e88183">

works on android. lemme fix the conflict and merge what I have :)